api/src/test/groovy/uk/gov/dhsc/htbhf/claimant/controller/NewClaimSpec.groovy

```diff
@@ -103,6 +88,26 @@ class NewClaimSpec extends Specification {
         "dateOfBirth" | "9999-12-31"  | "must be a past date"                      | "claimant.dateOfBirth"
     }
 
+    @Unroll
+    def "Field [#fieldName] with invalid value [#value] on an address returns the correct error response"(String fieldName, String value, String expectedErrorMessage, String expectedField) {
+        expect:
+        def claim = createClaimWithAddressProperty(fieldName, value)
+        def response = restTemplate.exchange(buildRequestEntity(claim), ErrorResponse.class)
+        assertValidationResponse(response, expectedField, expectedErrorMessage)
+
+        where:
+        fieldName      | value       | expectedErrorMessage             | expectedField
+        "addressLine1" | null        | "must not be null"               | "claimant.cardDeliveryAddress.addressLine1"
+        "addressLine1" | LONG_STRING | "size must be between 1 and 500" | "claimant.cardDeliveryAddress.addressLine1"
+        "addressLine2" | LONG_STRING | "size must be between 0 and 500" | "claimant.cardDeliveryAddress.addressLine2"
+        "townOrCity"   | null        | "must not be null"               | "claimant.cardDeliveryAddress.townOrCity"
+        "townOrCity"   | LONG_STRING | "size must be between 1 and 500" | "claimant.cardDeliveryAddress.townOrCity"
+        "postcode"     | "AA1122BB"  | "invalid postcode format"        | "claimant.cardDeliveryAddress.postcode"
+        "postcode"     | "A"         | "invalid postcode format"        | "claimant.cardDeliveryAddress.postcode"
+        "postcode"     | "11AA21"    | "invalid postcode format"        | "claimant.cardDeliveryAddress.postcode"
+        "postcode"     | ""          | "invalid postcode format"        | "claimant.cardDeliveryAddress.postcode"
```

Comment:
Can we have a test (probably needs to be separate from this test) for a null postcode please?